Watchman.Common/FriendlyDictionary.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Watchman.Common
+{
+    public class FriendlyDictionary<T, V>
+        where V : class, new()
+    {
+        private readonly IDictionary<T, V> _dictionary;
+
+        public FriendlyDictionary()
+        {
+            this._dictionary = new Dictionary<T, V>();
+        }
+
+        public FriendlyDictionary(IDictionary<T, V> dictionary)
+        {
+            this._dictionary = dictionary;
+        }
+
+        public bool ContainsKey(T key)
+        {
+            return this._dictionary.ContainsKey(key);
+        }
+
+        public V GetOrCreate(T key)
+        {
+            if (!this._dictionary.TryGetValue(key, out var value))
+            {
+                this.Set(key, new V());
```

Comment:
tak się zastanawiam, czy tutaj aby na pewno jest sens zawsze tworzyć taką komórkę jeśli nie znalazł tego klucza
może lepiej po prostu zwrócić pustą wartość?
po prostu czy ten słownik nie urośnie do wielkich rozmiarów gdzie większość to będą puste komórki?

Reply:
widzę że gdzieś używasz .Add na tak otrzymanej komórce, pewnie dlatego tak zrobiłeś
ale może da się jakoś inaczej?

Reply:
Mogę zrobić funkcję która przegląda dictionary i usuwa puste kontenery i używać jej np przy okazji wywoływania komendy usuwania warnów

Reply:
spoko

Reply:
gotowe